norduser/utmp.go

```diff
@@ -49,15 +40,8 @@ int get_utmp_user_processes(char*** users) {
       size++;
     }
 
-    (*users)[index] = malloc(__UT_NAMESIZE + 1);
-    if ((*users)[index] == NULL) {
-      free_users_table(users, size);
-      endutxent();
-      return ERROR_MALLOC_USERNAME;
-    }
-    (*users)[index][__UT_NAMESIZE]='\0';
-
-    strncpy((*users)[index], u->ut_user, __UT_NAMESIZE);
+	(*users)[index].login_pid = u->ut_pid;
+    strncpy((*users)[index].username, u->ut_user, __UT_NAMESIZE + 1);
```

Comment:
I think this will still be problematic if the `u->ut_user` has `__UT_NAMESIZE` chars, because it will take the byte after `u->ut_user`. And if the next byte is not `0` by accident, then `(*users)[index].username` will contain some garbage char and no `\0`.